lib/parallel_split_test/command_line.rb

```diff
@@ -41,7 +41,9 @@ def run(err, out)
 
     # modify + reparse args to unify output
     def modify_out_file_in_args(process_number)
-      @args[out_file_position] = process_number.to_s << "-" << out_file
+      target = out_file.split(".")
+      target[target.length - 2] = target[target.length - 2] << "." << process_number.to_s
```

Comment:
can you wrap this logic into a method and give it a name/some docs ... maybe File.extname() or so helps ...